grammar.y

```diff
@@ -1184,9 +1239,43 @@ table_constraint:
   }
 ;
 
+indexed_column_list:
+  indexed_column
+  {
+    $$ = IndexedColumnList{$1}
+  }
+| indexed_column_list ',' indexed_column
+  {
+    $$ = append($1, $3)
+  }
+;
+
+indexed_column:
+  column_name collate_opt primary_key_order
+  {
+    $$ = &IndexedColumn{Column : $1, CollationName: $2, Order: $3}
+  }
+;
+
+collate_opt:
+  {
+    $$ = Identifier("")
+  }
+| COLLATE identifier
+  {
+    $$ = Identifier(string($2))
+  }
+;
+
 insert_stmt:
   INSERT INTO table_name column_name_list_opt VALUES insert_rows upsert_clause_opt
   {
+    for i := 0; i < len($4); i++ {
```

Comment:
Check the columns of an `INSERT` and block references to `ROWID`

Reply:
Cool, that's useful in the case users provide the names at least :+1: